Geometry/HGCalCommonData/src/HGCalDDDConstants.cc

```diff
@@ -691,6 +695,116 @@ std::pair<float,float> HGCalDDDConstants::locateCellTrap(int lay, int irad,
   return std::make_pair(x,y);
 }
 
+bool HGCalDDDConstants::maskCell(const DetId& detId, int corners) const {
+  bool mask(false);
+  if (corners > 2 && corners < 6) {
+    if ((mode_ == HGCalGeometryMode::Hexagon8) ||
+	(mode_ == HGCalGeometryMode::Hexagon8Full)) {
+      int N(0), layer(0), waferU(0), waferV(0), u(0), v(0);
+      if (detId.det() == DetId::Forward) {
+	HFNoseDetId id(detId); 
+	N      = getUVMax(id.type());
+	layer  = id.layer();
+	waferU = id.waferU();
+	waferV = id.waferV();
+	u      = id.cellU();
+	v      = id.cellV();
+      } else {
+	HGCSiliconDetId id(detId);
```

Comment:
HFNoseDetId and HGCSiliconDetId have six methods that are the same. They should be derived from a base class that declares these methods, possibly as virtual.

Reply:
Please do not ask to modify DetId classes in this PR. That can be handled separately

Reply:
OK. I added an issue:
https://github.com/cms-sw/cmssw/issues/25062
This issue can be addressed in a later PR.